twill-yarn/src/main/java/org/apache/twill/yarn/YarnTwillController.java

```diff
@@ -77,7 +77,7 @@
    */
   YarnTwillController(String appName, RunId runId, ZKClient zkClient,
                       final ApplicationMasterLiveNodeData amLiveNodeData, final YarnAppClient yarnAppClient) {
-    super(runId, zkClient, Collections.<LogHandler>emptyList());
+    super(appName, runId, zkClient, amLiveNodeData.getKafkaZKConnect() != null, Collections.<LogHandler>emptyList());
```

Comment:
The contract of the getKafkaZKConnect won't change though, since it has to return something (or throw exception). From the caller point of view, there is almost no differences.